recipes/catkin/meta.yaml

```
@@ -0,0 +1,60 @@
+package:
+  name: ros-catkin
+  version: 0.7.17
+
+source:
+  - url: https://github.com/ros-gbp/catkin-release/archive/release/melodic/catkin/0.7.17-0.tar.gz
+    fn: ros-catkin-0.7.17.tar.gz
+    sha256: 316178707bc3f0edcb86a77ec2ec174b98b4d2ccf41306162cefe6588b3a9d41
+    patches:
+      - 001_remove_rt.patch
+      - 002_no_hardcoded_python.patch
+  - path: 003_ros_setup.patch
+  - path: 004_setup_util.patch
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+  host:
+    - ros-conda-mutex * melodic
+    - ros-conda-base
+    - python
+    - catkin_pkg
+    - empy
+    - cmake
+  run:
+    - ros-conda-mutex * melodic
+    - ros-conda-base
+    - gtest
+    - nose
+    - empy
+    - gmock
+
+test:
+  requires:
+    - rospkg
+  commands:
+    - catkin_find
+    - catkin_find --help
+    - catkin_make --help
+    - export ROS_ROOT=$PREFIX && python -c "import rospkg;s = rospkg.RosPack();ret = s.get_path('catkin');print(ret)"
+
+about:
+  home: http://www.ros.org/wiki/catkin
+  summary: Low-level build system macros and infrastructure for ROS.
+  description: |
+    Low-level build system macros and infrastructure for ROS.
+  license: BSD
```

Which licence is it? (Please check I guessed correctly!)
```suggestion
  license: BSD-3-Clause
  license_family: BSD
```

---

this is extracted from the `package.xml` file of the ROS package. https://github.com/ros/catkin/blob/461cd3c217f952f908a3cba56927660764b9ae62/package.xml#L10

I would really dislike to change this. Also the package.xml doesn't know about license family so I don't really want to add that.

---

I would push to fix the `license` field at least as having consistent metadata across the entirety of conda-forge is useful for black/white-listing licences. That said I won't push any further as the metadata still has a long way to go before it's consistent 😄

(Can you parse the licence file to figure out which variant it is?)

---

hmm, i found scancode (https://github.com/nexB/scancode-toolkit) which might do what we need. I can try to integrate it into the recipe-generation toolchain (also, because a lot of the ROS packages don't ship with a LICENSE file).

---

scancode-toolkit seems to work :) I'll try to integrate that in the pipeline

---

That's much better than I was thinking and looks very useful (🤔...conda-skeleton)